include/core/CStreamWriter.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the following additional limitation. Functionality enabled by the
+ * files subject to the Elastic License 2.0 may only be used in production when
+ * invoked by an Elasticsearch process with a license key installed that permits
+ * use of machine learning features. You may not use this file except in
+ * compliance with the Elastic License 2.0 and the foregoing additional
+ * limitation.
+ */
+
+#ifndef INCLUDED_ml_core_CStreamWriter_h
+#define INCLUDED_ml_core_CStreamWriter_h
+
+#include <core/CBoostJsonLineWriter.h>
+
+#include <boost/json.hpp>
+
+namespace ml {
+namespace core {
+using TBoostJsonLineWriterStream = core::CBoostJsonLineWriter<std::ostream>;
+
+class CStreamWriter : public TBoostJsonLineWriterStream {
+public:
+    CStreamWriter(std::ostream& os) : TBoostJsonLineWriterStream(os) {}
+    CStreamWriter() : TBoostJsonLineWriterStream() {}
+
+    void append(const std::string_view& str) override {
+        (*m_Os) << str;
+        m_DebugBuffer.append(str);
```

Comment:
Are we using this class in production? If so, it seems like it will cause problems to build up the entire output in a string. It will be fine in tests but then will blow up at a customer with a 20GB model state.

Alternatively, if this class is _only_ for use in tests, please add a Doxygen comment to say this. Also, could it be moved to the `test` library if that's not too complicated?

Reply:
Removed - it was only ever useful during development.